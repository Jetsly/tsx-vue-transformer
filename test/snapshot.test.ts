import * as ts from 'typescript';
import { join } from 'path';
import { readdirSync, statSync, readFileSync } from 'fs';
import transformerFactory from '../src';
const printer = ts.createPrinter();
const FIXTURES_DIR = join(__dirname, '__fixtures__');

describe('snapshot', () => {
  readdirSync(FIXTURES_DIR).forEach(fixtureName => {
    const fixtureFile = join(FIXTURES_DIR, fixtureName);
    if (statSync(fixtureFile).isFile()) {
      it(fixtureName.split('-').join(' '), () => {
        const sourceCode = readFileSync(fixtureFile, 'utf-8');
        const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
        const result = ts.transform(source, [transformerFactory()]);
        const transformedSourceFile = result.transformed[0];
        const resultCode = printer.printFile(transformedSourceFile);
        expect(resultCode).toMatchSnapshot();
      });
    }
  });
});
