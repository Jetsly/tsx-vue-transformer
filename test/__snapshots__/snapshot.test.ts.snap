// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot functional.tsx 1`] = `
"export const text = 'foo';
export const noop = _ => _;
export const ContainsText = {
    render(h) {
        return h(\\"div\\", [\\"test\\"]);
    },
};
export const BindsText = {
    render(h) {
        return h(\\"div\\", [text]);
    },
};
export const ExtractsAttrs = {
    render(h) {
        return h(\\"div\\", { attrs: { \\"id\\": \\"hi\\", \\"dir\\": \\"ltr\\" } });
    },
};
export const BindsAttrs = {
    render(h) {
        return h(\\"div\\", { attrs: { \\"id\\": text } });
    },
};
export const HandlesTopLevelSpecialAttrs = {
    render(h) {
        return h(\\"div\\", { \\"class\\": text, \\"style\\": \\"bar\\", \\"key\\": \\"key\\", \\"ref\\": \\"ref\\", \\"refInFor\\": true, \\"slot\\": \\"slot\\" });
    },
};
export const HandlesNestedProperties = {
    render(h) {
        return (h(\\"div\\", { props: { \\"on-success\\": noop }, on: { \\"click\\": noop, \\"kebab-case\\": noop }, domProps: { \\"innerHTML\\": \\"<p>hi</p>\\" }, hook: { \\"insert\\": noop } }));
    },
};
"
`;
